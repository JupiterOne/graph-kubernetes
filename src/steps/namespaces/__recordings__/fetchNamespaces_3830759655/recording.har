{
  "log": {
    "_recordingName": "fetchNamespaces",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "e262abbd5b6acc4c1139793afa880ed2",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json"
            },
            {
              "name": "host",
              "value": "192.168.49.2:8443"
            }
          ],
          "headersSize": 115,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "250"
            }
          ],
          "url": "https://192.168.49.2:8443/api/v1/namespaces?limit=250"
        },
        "response": {
          "bodySize": 2052,
          "content": {
            "mimeType": "application/json",
            "size": 2052,
            "text": "{\"kind\":\"NamespaceList\",\"apiVersion\":\"v1\",\"metadata\":{\"resourceVersion\":\"51782\"},\"items\":[{\"metadata\":{\"name\":\"default\",\"uid\":\"868d4703-b338-4051-98e4-b65b4f0a0006\",\"resourceVersion\":\"39\",\"creationTimestamp\":\"2024-01-11T15:50:15Z\",\"labels\":{\"kubernetes.io/metadata.name\":\"default\"},\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2024-01-11T15:50:15Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:metadata\":{\"f:labels\":{\".\":{},\"f:kubernetes.io/metadata.name\":{}}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-node-lease\",\"uid\":\"52615a92-75c3-4c4b-8b72-51be82a4bc21\",\"resourceVersion\":\"35\",\"creationTimestamp\":\"2024-01-11T15:50:15Z\",\"labels\":{\"kubernetes.io/metadata.name\":\"kube-node-lease\"},\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2024-01-11T15:50:15Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:metadata\":{\"f:labels\":{\".\":{},\"f:kubernetes.io/metadata.name\":{}}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-public\",\"uid\":\"7a2ee4f7-12fd-494b-93f0-c639b1e63a72\",\"resourceVersion\":\"20\",\"creationTimestamp\":\"2024-01-11T15:50:15Z\",\"labels\":{\"kubernetes.io/metadata.name\":\"kube-public\"},\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2024-01-11T15:50:15Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:metadata\":{\"f:labels\":{\".\":{},\"f:kubernetes.io/metadata.name\":{}}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-system\",\"uid\":\"ce86116b-863b-4e74-979e-3c031a9aeee1\",\"resourceVersion\":\"12\",\"creationTimestamp\":\"2024-01-11T15:50:15Z\",\"labels\":{\"kubernetes.io/metadata.name\":\"kube-system\"},\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2024-01-11T15:50:15Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:metadata\":{\"f:labels\":{\".\":{},\"f:kubernetes.io/metadata.name\":{}}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "audit-id",
              "value": "9257c42e-279b-4392-bd42-e673be33566e"
            },
            {
              "name": "cache-control",
              "value": "no-cache, private"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "x-kubernetes-pf-flowschema-uid",
              "value": "3e535193-da03-4be0-8af9-31b77c23de3e"
            },
            {
              "name": "x-kubernetes-pf-prioritylevel-uid",
              "value": "1a41d4ba-2c5d-4828-b199-fd4ff13a2398"
            },
            {
              "name": "date",
              "value": "Fri, 12 Jan 2024 17:13:39 GMT"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            }
          ],
          "headersSize": 343,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-01-12T17:13:39.650Z",
        "time": 22,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 22
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
