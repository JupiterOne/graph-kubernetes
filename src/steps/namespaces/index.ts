import { IntegrationStep } from '@jupiterone/integration-sdk-core';
import { IntegrationConfig, IntegrationStepContext } from '../../config';
import { Entities, IntegrationSteps } from '../constants';
import { createNamespaceEntity } from './converters';
import getOrCreateAPIClient from '../../kubernetes/getOrCreateAPIClient';

export async function fetchNamespaces(
  context: IntegrationStepContext,
): Promise<void> {
  const { instance, jobState } = context;
  const { config } = instance;

  const client = getOrCreateAPIClient(config);

  await client.iterateNamespaces(async (namespace) => {
    const namespaceEntity = createNamespaceEntity(namespace);
    await jobState.addEntity(namespaceEntity);
  });
}

export const namespaceSteps: IntegrationStep<IntegrationConfig>[] = [
  {
    id: IntegrationSteps.NAMESPACES,
    name: 'Fetch Namespaces',
    entities: [Entities.NAMESPACE],
    relationships: [],
    dependsOn: [],
    executionHandler: fetchNamespaces,
  },
];
