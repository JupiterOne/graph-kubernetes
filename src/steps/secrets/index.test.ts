import { fetchSecrets } from '.';
import { fetchNamespaces } from '../namespaces';
import { createDataCollectionTest } from '../../../test/recording';
import { integrationConfig } from '../../../test/config';
import { Entities, Relationships } from '../constants';
import { RelationshipClass } from '@jupiterone/integration-sdk-core';

describe('#fetchSecrets', () => {
  test('should collect data', async () => {
    await createDataCollectionTest({
      recordingName: 'fetchSecrets',
      recordingDirectory: __dirname,
      integrationConfig,
      stepFunctions: [fetchNamespaces, fetchSecrets],
      entitySchemaMatchers: [
        {
          _type: Entities.SECRET._type,
          matcher: {
            _class: ['Vault'],
            schema: {
              additionalProperties: false,
              properties: {
                _type: { const: 'kube_secret' },
                _rawData: {
                  type: 'array',
                  items: { type: 'object' },
                },
                name: { type: 'string' },
                displayName: { type: 'string' },
                generation: { type: 'number' },
                deletionGracePeriodSeconds: { type: 'string' },
                resourceVersion: { type: 'string' },
                createdOn: { type: 'number' },
                deletedOn: { type: 'number' },
                type: { type: 'string' },
                classification: { const: null },
                encrypted: { type: 'boolean' },
              },
            },
          },
        },
      ],
      relationshipSchemaMatchers: [
        {
          _type: Relationships.NAMESPACE_CONTAINS_SECRET._type,
          matcher: {
            schema: {
              properties: {
                _class: { const: RelationshipClass.CONTAINS },
                _type: { const: 'kube_namespace_contains_secret' },
              },
            },
          },
        },
      ],
    });
  });
});
