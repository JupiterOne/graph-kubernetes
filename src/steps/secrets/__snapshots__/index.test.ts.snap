// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchSecrets should collect data: jobState 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "868d4703-b338-4051-98e4-b65b4f0a0006",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "creationTimestamp": 2024-01-11T15:50:15.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": Object {
                "kubernetes.io/metadata.name": "default",
              },
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:metadata": Object {
                      "f:labels": Object {
                        ".": Object {},
                        "f:kubernetes.io/metadata.name": Object {},
                      },
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "subresource": undefined,
                  "time": 2024-01-11T15:50:15.000Z,
                },
              ],
              "name": "default",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "39",
              "selfLink": undefined,
              "uid": "868d4703-b338-4051-98e4-b65b4f0a0006",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1704988215000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "default",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "default",
      "resourceVersion": "39",
      "status.phase": "Active",
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "52615a92-75c3-4c4b-8b72-51be82a4bc21",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "creationTimestamp": 2024-01-11T15:50:15.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": Object {
                "kubernetes.io/metadata.name": "kube-node-lease",
              },
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:metadata": Object {
                      "f:labels": Object {
                        ".": Object {},
                        "f:kubernetes.io/metadata.name": Object {},
                      },
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "subresource": undefined,
                  "time": 2024-01-11T15:50:15.000Z,
                },
              ],
              "name": "kube-node-lease",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "35",
              "selfLink": undefined,
              "uid": "52615a92-75c3-4c4b-8b72-51be82a4bc21",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1704988215000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "kube-node-lease",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "kube-node-lease",
      "resourceVersion": "35",
      "status.phase": "Active",
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "7a2ee4f7-12fd-494b-93f0-c639b1e63a72",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "creationTimestamp": 2024-01-11T15:50:15.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": Object {
                "kubernetes.io/metadata.name": "kube-public",
              },
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:metadata": Object {
                      "f:labels": Object {
                        ".": Object {},
                        "f:kubernetes.io/metadata.name": Object {},
                      },
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "subresource": undefined,
                  "time": 2024-01-11T15:50:15.000Z,
                },
              ],
              "name": "kube-public",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "20",
              "selfLink": undefined,
              "uid": "7a2ee4f7-12fd-494b-93f0-c639b1e63a72",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1704988215000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "kube-public",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "kube-public",
      "resourceVersion": "20",
      "status.phase": "Active",
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "ce86116b-863b-4e74-979e-3c031a9aeee1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "creationTimestamp": 2024-01-11T15:50:15.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": Object {
                "kubernetes.io/metadata.name": "kube-system",
              },
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:metadata": Object {
                      "f:labels": Object {
                        ".": Object {},
                        "f:kubernetes.io/metadata.name": Object {},
                      },
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "subresource": undefined,
                  "time": 2024-01-11T15:50:15.000Z,
                },
              ],
              "name": "kube-system",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "12",
              "selfLink": undefined,
              "uid": "ce86116b-863b-4e74-979e-3c031a9aeee1",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1704988215000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "kube-system",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "kube-system",
      "resourceVersion": "12",
      "status.phase": "Active",
    },
    Object {
      "_class": Array [
        "Vault",
      ],
      "_key": "bd3b80ae-ae33-4437-94ce-10a31e912bdc",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "apiVersion": undefined,
            "immutable": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": Object {
                "kubectl.kubernetes.io/last-applied-configuration": "{\\"apiVersion\\":\\"v1\\",\\"data\\":{\\"accessType\\":\\"Y2x1c3Rlcg==\\",\\"api_key\\":\\"IjgzYTcxZjlkYTYyN2E3MDE5MjA5MWE2MzQwYzUwYjY2MjgwZjM2NzU4OGQyOTQzYWVkMzEzNzMwMzQzODMzMzUzMzMzMzAzNjMyMzYi\\",\\"jupiteroneAccountId\\":\\"ajFkZXY=\\",\\"jupiteroneApiKey\\":\\"OWMxZGRjOGJmMDBhMWYzZjQxZmY0MWZhYzQ5ZDI3MmRhNDBmNjMwOWRiNjVmODZjM2QzMTM3MzAzNDM3MzUzMDMyMzQzNjM2MzAzMw==\\",\\"jupiteroneIntegrationInstanceId\\":\\"NTZlNmQwZjItYTY5Mi00MGQ3LWEyN2QtOGY4YzRlOWVmOWFk\\",\\"loadKubernetesConfigFromDefault\\":\\"dHJ1ZQ==\\",\\"namespace\\":\\"ZGVmYXVsdA==\\"},\\"kind\\":\\"Secret\\",\\"metadata\\":{\\"annotations\\":{},\\"labels\\":{\\"app.kubernetes.io/managed-by\\":\\"tilt\\"},\\"name\\":\\"graph-kubernetes\\",\\"namespace\\":\\"default\\"}}
",
              },
              "creationTimestamp": 2024-01-12T16:59:26.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": Object {
                "app.kubernetes.io/managed-by": "tilt",
              },
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:data": Object {
                      ".": Object {},
                      "f:accessType": Object {},
                      "f:api_key": Object {},
                      "f:jupiteroneAccountId": Object {},
                      "f:jupiteroneApiKey": Object {},
                      "f:jupiteroneIntegrationInstanceId": Object {},
                      "f:loadKubernetesConfigFromDefault": Object {},
                      "f:namespace": Object {},
                    },
                    "f:metadata": Object {
                      "f:annotations": Object {
                        ".": Object {},
                        "f:kubectl.kubernetes.io/last-applied-configuration": Object {},
                      },
                      "f:labels": Object {
                        ".": Object {},
                        "f:app.kubernetes.io/managed-by": Object {},
                      },
                    },
                    "f:type": Object {},
                  },
                  "manager": "Go-http-client",
                  "operation": "Update",
                  "subresource": undefined,
                  "time": 2024-01-12T16:59:26.000Z,
                },
              ],
              "name": "graph-kubernetes",
              "namespace": "default",
              "ownerReferences": undefined,
              "resourceVersion": "50624",
              "selfLink": undefined,
              "uid": "bd3b80ae-ae33-4437-94ce-10a31e912bdc",
            },
            "stringData": undefined,
            "type": "Opaque",
          },
        },
      ],
      "_type": "kube_secret",
      "classification": null,
      "createdOn": 1705078766000,
      "deletedOn": undefined,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "graph-kubernetes",
      "encrypted": false,
      "generation": undefined,
      "name": "graph-kubernetes",
      "resourceVersion": "50624",
      "type": "Opaque",
    },
    Object {
      "_class": Array [
        "Vault",
      ],
      "_key": "f58b966c-2a2b-4314-82a1-580390097378",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "apiVersion": undefined,
            "immutable": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": Object {
                "meta.helm.sh/release-name": "np-test",
                "meta.helm.sh/release-namespace": "default",
              },
              "creationTimestamp": 2024-01-11T21:52:37.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": Object {
                "app.kubernetes.io/instance": "np-test",
                "app.kubernetes.io/managed-by": "Helm",
                "app.kubernetes.io/name": "mariadb",
                "app.kubernetes.io/version": "11.2.2",
                "helm.sh/chart": "mariadb-15.0.1",
              },
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:data": Object {
                      ".": Object {},
                      "f:mariadb-password": Object {},
                      "f:mariadb-root-password": Object {},
                    },
                    "f:metadata": Object {
                      "f:annotations": Object {
                        ".": Object {},
                        "f:meta.helm.sh/release-name": Object {},
                        "f:meta.helm.sh/release-namespace": Object {},
                      },
                      "f:labels": Object {
                        ".": Object {},
                        "f:app.kubernetes.io/instance": Object {},
                        "f:app.kubernetes.io/managed-by": Object {},
                        "f:app.kubernetes.io/name": Object {},
                        "f:app.kubernetes.io/version": Object {},
                        "f:helm.sh/chart": Object {},
                      },
                    },
                    "f:type": Object {},
                  },
                  "manager": "helm",
                  "operation": "Update",
                  "subresource": undefined,
                  "time": 2024-01-11T21:52:37.000Z,
                },
              ],
              "name": "np-test-mariadb",
              "namespace": "default",
              "ownerReferences": undefined,
              "resourceVersion": "27433",
              "selfLink": undefined,
              "uid": "f58b966c-2a2b-4314-82a1-580390097378",
            },
            "stringData": undefined,
            "type": "Opaque",
          },
        },
      ],
      "_type": "kube_secret",
      "classification": null,
      "createdOn": 1705009957000,
      "deletedOn": undefined,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "np-test-mariadb",
      "encrypted": false,
      "generation": undefined,
      "name": "np-test-mariadb",
      "resourceVersion": "27433",
      "type": "Opaque",
    },
    Object {
      "_class": Array [
        "Vault",
      ],
      "_key": "f34bcb0e-730f-476b-bd58-225abadc32a1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "apiVersion": undefined,
            "immutable": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": Object {
                "meta.helm.sh/release-name": "np-test",
                "meta.helm.sh/release-namespace": "default",
              },
              "creationTimestamp": 2024-01-11T21:52:37.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": Object {
                "app.kubernetes.io/instance": "np-test",
                "app.kubernetes.io/managed-by": "Helm",
                "app.kubernetes.io/name": "wordpress",
                "app.kubernetes.io/version": "6.4.2",
                "helm.sh/chart": "wordpress-19.0.5",
              },
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:data": Object {
                      ".": Object {},
                      "f:wordpress-password": Object {},
                    },
                    "f:metadata": Object {
                      "f:annotations": Object {
                        ".": Object {},
                        "f:meta.helm.sh/release-name": Object {},
                        "f:meta.helm.sh/release-namespace": Object {},
                      },
                      "f:labels": Object {
                        ".": Object {},
                        "f:app.kubernetes.io/instance": Object {},
                        "f:app.kubernetes.io/managed-by": Object {},
                        "f:app.kubernetes.io/name": Object {},
                        "f:app.kubernetes.io/version": Object {},
                        "f:helm.sh/chart": Object {},
                      },
                    },
                    "f:type": Object {},
                  },
                  "manager": "helm",
                  "operation": "Update",
                  "subresource": undefined,
                  "time": 2024-01-11T21:52:37.000Z,
                },
              ],
              "name": "np-test-wordpress",
              "namespace": "default",
              "ownerReferences": undefined,
              "resourceVersion": "27432",
              "selfLink": undefined,
              "uid": "f34bcb0e-730f-476b-bd58-225abadc32a1",
            },
            "stringData": undefined,
            "type": "Opaque",
          },
        },
      ],
      "_type": "kube_secret",
      "classification": null,
      "createdOn": 1705009957000,
      "deletedOn": undefined,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "np-test-wordpress",
      "encrypted": false,
      "generation": undefined,
      "name": "np-test-wordpress",
      "resourceVersion": "27432",
      "type": "Opaque",
    },
    Object {
      "_class": Array [
        "Vault",
      ],
      "_key": "44660625-d335-4529-85e0-b6727dde78ba",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "apiVersion": undefined,
            "immutable": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "creationTimestamp": 2024-01-11T21:52:37.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": Object {
                "modifiedAt": "1705009957",
                "name": "np-test",
                "owner": "helm",
                "status": "deployed",
                "version": "1",
              },
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:data": Object {
                      ".": Object {},
                      "f:release": Object {},
                    },
                    "f:metadata": Object {
                      "f:labels": Object {
                        ".": Object {},
                        "f:modifiedAt": Object {},
                        "f:name": Object {},
                        "f:owner": Object {},
                        "f:status": Object {},
                        "f:version": Object {},
                      },
                    },
                    "f:type": Object {},
                  },
                  "manager": "Helm",
                  "operation": "Update",
                  "subresource": undefined,
                  "time": 2024-01-11T21:52:37.000Z,
                },
              ],
              "name": "sh.helm.release.v1.np-test.v1",
              "namespace": "default",
              "ownerReferences": undefined,
              "resourceVersion": "27473",
              "selfLink": undefined,
              "uid": "44660625-d335-4529-85e0-b6727dde78ba",
            },
            "stringData": undefined,
            "type": "helm.sh/release.v1",
          },
        },
      ],
      "_type": "kube_secret",
      "classification": null,
      "createdOn": 1705009957000,
      "deletedOn": undefined,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "sh.helm.release.v1.np-test.v1",
      "encrypted": false,
      "generation": undefined,
      "name": "sh.helm.release.v1.np-test.v1",
      "resourceVersion": "27473",
      "type": "helm.sh/release.v1",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "CONTAINS",
      "_fromEntityKey": "868d4703-b338-4051-98e4-b65b4f0a0006",
      "_key": "868d4703-b338-4051-98e4-b65b4f0a0006|contains|bd3b80ae-ae33-4437-94ce-10a31e912bdc",
      "_toEntityKey": "bd3b80ae-ae33-4437-94ce-10a31e912bdc",
      "_type": "kube_namespace_contains_secret",
      "displayName": "CONTAINS",
    },
    Object {
      "_class": "CONTAINS",
      "_fromEntityKey": "868d4703-b338-4051-98e4-b65b4f0a0006",
      "_key": "868d4703-b338-4051-98e4-b65b4f0a0006|contains|f58b966c-2a2b-4314-82a1-580390097378",
      "_toEntityKey": "f58b966c-2a2b-4314-82a1-580390097378",
      "_type": "kube_namespace_contains_secret",
      "displayName": "CONTAINS",
    },
    Object {
      "_class": "CONTAINS",
      "_fromEntityKey": "868d4703-b338-4051-98e4-b65b4f0a0006",
      "_key": "868d4703-b338-4051-98e4-b65b4f0a0006|contains|f34bcb0e-730f-476b-bd58-225abadc32a1",
      "_toEntityKey": "f34bcb0e-730f-476b-bd58-225abadc32a1",
      "_type": "kube_namespace_contains_secret",
      "displayName": "CONTAINS",
    },
    Object {
      "_class": "CONTAINS",
      "_fromEntityKey": "868d4703-b338-4051-98e4-b65b4f0a0006",
      "_key": "868d4703-b338-4051-98e4-b65b4f0a0006|contains|44660625-d335-4529-85e0-b6727dde78ba",
      "_toEntityKey": "44660625-d335-4529-85e0-b6727dde78ba",
      "_type": "kube_namespace_contains_secret",
      "displayName": "CONTAINS",
    },
  ],
  "encounteredTypes": Array [
    "kube_namespace",
    "kube_secret",
    "kube_namespace_contains_secret",
  ],
  "numCollectedEntities": 8,
  "numCollectedRelationships": 4,
}
`;
